macOSUpgrade.sh

```diff
@@ -374,12 +378,16 @@ if [[ ${pwrStatus} == "OK" ]] && [[ ${spaceStatus} == "OK" ]]; then
         eraseopt='--eraseinstall'
         /bin/echo "   Script is configured for Erase and Install of macOS."
     fi
+    ##Check if installPackage is Enabled
+    if [ ! -z "${installPkg}" ]; then
+        pkgopt="--installpackage ${installPkg}"
```

Comment:
@taniguti 

There is something I would like to discuss with you.

I would say @ecanault's work has partially good points. 
But it might be better to use following steps than to use the `installpackage` option of `startosinstall` command.

**Steps (draft)**

1. Get the trigger name of a policy in the argument from the user with jamf script argument.
2. Create a pkg that runs `jamf policy-event <the trigger name>`.
3. The value of the `installpackage` option should be set to run the previously created pkg. 

This is because it is difficult to put pkg on the local device beforehand.
The advantage of running a policy is that we can run a variety of subsequent processes, not just a pkg.

Reply:
@kenchan0130 

Summary)
The addition of pkgopt complicates the functionality of macOSUpgrade.sh, so I object to adding this functionality.
I think that it is possible to realize this function by the function of JamfPro.

----
It is of course possible to use macOSUpgrade.sh alone, but I understand that it is assumed to be used with JamfPro.
This means that the Mac computer on which macOSUpgrade.sh is to be run is under control of JamfPro.

Considering putting the pkg file in the path pointed to by "$ pkgotp", it seems more straightforward to do with JamfPro's Policy processing than processing in macOSUpgrade.sh. In other words, check the version of os etc. If there is no element added by the pkg file, pkg will be sent and installed.

Even through, when adding "pkgopt" to macOSUpgrade.sh, there should be confirmation of the existence of the pkg file expected by this switch. There should also be  verification of the package itself, if it might not be signed or the file has been corrupted. The implicit premise and expectation that management will definitely put the pkg file in its place as expected brings the possibility of interfering with stable operation.
It is possible to make the above implementation macOSUpgrade.sh, but the code gets more complicated. I think it would be better to adopt a solution using JamfPro's Policy.

Thank you for your suggestions. @ecanault

Reply:
> The addition of pkgopt complicates the functionality of macOSUpgrade.sh, so I object to adding this functionality.
> I think that it is possible to realize this function by the function of JamfPro.

I have the same opinion as you.
As mentioned https://github.com/kc9wwh/macOSUpgrade/pull/110#issuecomment-495802023
, I recommend to use the install pkg feature of prestage enrollment if the admin want to use erase option of the `startosinstall` command.

However, when the device does not have DEP, the `installpackage` option may be useful.
Therefore, if package validation is correct, we may be able to accept the changes.
We will not continue to adopt it aggressively, as it will not be complicated. (In other words, if we get the right response, we may adopt it.)